File: test_cases/javascript/function_made_generator/before.js
// Before: Regular functions
function createSequence(start, end) {
    const result = [];
    for (let i = start; i <= end; i++) {
        result.push(i);
    }
    return result;
}

function fibonacci(n) {
    if (n <= 1) return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

function processNumbers(numbers) {
    const results = [];
    for (let num of numbers) {
        results.push(num * 2);
    }
    return results;
}

File: test_cases/python/function_made_generator/before.py
def fetch_data(url):
    """Synchronous data fetching."""
    # Simulate network request
    import time
    time.sleep(0.1)
    return f"Data from {url}"

def process_items(items):
    """Synchronous item processing."""
    results = []
    for item in items:
        # Process each item
        result = item * 2
        results.append(result)
    return results

def generate_numbers(n):
    """Regular function that returns a list."""
    numbers = []
    for i in range(n):
        numbers.append(i * i)
    return numbers
